src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -33368,635 +33971,3631 @@ .maxstack  3
+                }
+                static class Extensions
+                {
+                    public static void Add<T>(this MyCollection<T> collection, {{refKind}} T t) { t = default; collection.__AddInternal(t); }
+                }
+                namespace N
+                {
+                    static class Extensions
+                    {
+                        public static void Add<T>(this MyCollection<T> collection, T t) { collection.__AddInternal(t); }
+                    }
+                }
+                class Program
+                {
+                    static void Main()
+                    {
+                        int x = 1;
+                        int[] y = [2, 3];
+                        MyCollection<int?> z = [x, ..y, null];
+                        z.Report();
+                    }
+                }
+                """;
+            CompileAndVerify([source, s_collectionExtensions], expectedOutput: "[1, 2, 3, null], ");
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("in")]
+        [InlineData("ref readonly")]
+        public void AddMethod_ByRef_Extension_06(string refKind)
+        {
+            string source = $$"""
+                using System.Collections;
+                using System.Collections.Generic;
+                interface IA { }
+                interface IB { }
+                interface IC { }
+                class C : IA, IB, IC
+                {
+                    private readonly int _i;
+                    public C(int i) { _i = i; }
+                    public override string ToString() => _i.ToString();
+                    public static implicit operator C(int i) => new(i);
+                }
+                class MyCollection<T> : IEnumerable
+                {
+                    private readonly List<object> _list = new();
+                    IEnumerator IEnumerable.GetEnumerator() => _list.GetEnumerator();
+                    public void Add(out IA a) => throw null;
+                    public void Add(ref IB b) => throw null;
+                    internal void __AddInternal(IC c) { _list.Add(c); }
+                }
+                static class Extensions
+                {
+                    public static void Add<T>(this MyCollection<T> collection, {{refKind}} IC c) { collection.__AddInternal(c); }
+                }
+                class Program
+                {
+                    static void Main()
+                    {
+                        C x = 1;
+                        C[] y = [2, 3];
+                        MyCollection<C> z = [x, ..y, null];
+                        z.Report();
+                    }
+                }
+                """;
+            CompileAndVerify([source, s_collectionExtensions], expectedOutput: "[1, 2, 3, null], ");
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("in")]
+        [InlineData("ref readonly")]
```

Comment:
Why does this work for `ref readonly`? I would, at the very least, expect a warning?

Reply:
Same response, I guess.